backend/src/schema/resolvers/notifications.spec.js

```diff
@@ -145,80 +243,201 @@ describe('given some notifications', () => {
+                    __typename: 'Report',
+                    id: 'reportOnUser',
+                    filed: [
+                      {
+                        reasonCategory: 'discrimination_etc',
+                        reasonDescription: 'This user is harassing me with bigoted remarks!',
+                        reportedResource: {
+                          __typename: 'User',
+                          id: 'badWomen',
+                          name: 'Mrs. Badwomen',
+                        },
+                      },
+                    ],
                   },
+                },
+                {
+                  createdAt: '2020-01-14T12:33:48.651Z',
+                  updatedAt: '2020-01-14T12:33:48.651Z',
                   read: false,
-                  createdAt: '2019-08-30T19:33:48.651Z',
+                  reason: 'filed_report_on_resource',
+                  from: {
+                    __typename: 'Report',
+                    id: 'reportOnComment',
+                    filed: [
+                      {
+                        reasonCategory: 'discrimination_etc',
+                        reasonDescription: 'This user is harassing me!',
+                        reportedResource: {
+                          __typename: 'Comment',
+                          id: 'c4',
+                          content:
+                            'I am harassing content in a harassing comment to a bad post !!!',
+                        },
+                      },
+                    ],
+                  },
                 },
                 {
+                  createdAt: '2019-08-31T17:33:48.651Z',
+                  updatedAt: '2019-08-31T17:33:48.651Z',
+                  read: false,
+                  reason: 'mentioned_in_post',
                   from: {
                     __typename: 'Post',
                     content: 'You have been mentioned in a post',
                   },
+                },
+                {
+                  createdAt: '2019-08-30T19:33:48.651Z',
+                  updatedAt: '2019-08-30T19:33:48.651Z',
                   read: false,
-                  createdAt: '2019-08-31T17:33:48.651Z',
+                  reason: 'mentioned_in_comment',
+                  from: {
+                    __typename: 'Comment',
+                    content: 'You have been mentioned in a comment',
+                  },
                 },
-              ]),
+              ],
             },
-          })
+            errors: undefined,
+          }
+
           const response = await query({
             query: notificationQuery,
             variables: { ...variables, read: false },
           })
           await expect(response).toMatchObject(expected)
-          await expect(response.data.notifications.length).toEqual(2) // double-check
+          await expect(response.data.notifications.length).toEqual(4) // has to be checked, because of 'arrayContaining'
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2020-01-23T19:01:29Z" title="Thursday, January 23rd 2020, 8:01:29 pm +01:00">Jan 23, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -145,80 +233,188 @@ describe('given some notifications', () => {
 
       describe('no filters', () => {
         it('returns all notifications of current user', async () => {
-          const expected = [
-            {
-              from: {
-                __typename: 'Comment',
-                content: 'You have seen this comment mentioning already',
-              },
-              read: true,
-              createdAt: '2019-08-30T15:33:48.651Z',
-            },
-            {
-              from: {
-                __typename: 'Post',
-                content: 'Already seen post mention',
-              },
-              read: true,
-              createdAt: '2019-08-30T17:33:48.651Z',
-            },
-            {
-              from: {
-                __typename: 'Comment',
-                content: 'You have been mentioned in a comment',
-              },
-              read: false,
-              createdAt: '2019-08-30T19:33:48.651Z',
-            },
-            {
-              from: {
-                __typename: 'Post',
-                content: 'You have been mentioned in a post',
-              },
-              read: false,
-              createdAt: '2019-08-31T17:33:48.651Z',
-            },
-          ]
-
-          await expect(query({ query: notificationQuery, variables })).resolves.toMatchObject({
+          const expected = {
             data: {
-              notifications: expect.arrayContaining(expected),
+              notifications: expect.arrayContaining([
+                expect.objectContaining({
+                  from: {
+                    __typename: 'Comment',
+                    content: 'You have seen this comment mentioning already',
+                  },
+                  read: true,
+                  createdAt: '2019-08-30T15:33:48.651Z',
+                }),
+                expect.objectContaining({
+                  from: {
+                    __typename: 'Post',
+                    content: 'Already seen post mention',
+                  },
+                  read: true,
+                  createdAt: '2019-08-30T17:33:48.651Z',
+                }),
+                expect.objectContaining({
+                  from: {
+                    __typename: 'Comment',
+                    content: 'You have been mentioned in a comment',
+                  },
+                  read: false,
+                  createdAt: '2019-08-30T19:33:48.651Z',
+                }),
+                expect.objectContaining({
+                  from: {
+                    __typename: 'Post',
+                    content: 'You have been mentioned in a post',
+                  },
+                  read: false,
+                  createdAt: '2019-08-31T17:33:48.651Z',
+                }),
+                {
+                  createdAt: '2020-01-15T16:33:48.651Z',
+                  read: false,
+                  reason: 'filed_report_on_resource',
+                  from: {
+                    __typename: 'Report',
+                    id: 'reportOnUser',
+                    filed: [
+                      {
+                        reasonCategory: 'discrimination_etc',
+                        reasonDescription: 'This user is harassing me with bigoted remarks!',
+                        reportedResource: {
+                          __typename: 'User',
+                          id: 'badWomen',
+                          name: 'Mrs. Badwomen',
+                        },
+                      },
+                    ],
+                  },
+                },
+                {
+                  createdAt: '2020-01-16T10:33:48.651Z',
+                  read: true,
+                  reason: 'filed_report_on_resource',
+                  from: {
+                    __typename: 'Report',
+                    id: 'reportOnPost',
+                    filed: [
+                      {
+                        reasonCategory: 'other',
+                        reasonDescription:
+                          "This shouldn't be shown to anybody else! It's my private thing!",
+                        reportedResource: {
+                          __typename: 'Post',
+                          id: 'p4',
+                          title: 'Bad Post',
+                          content: 'I am bad content !!!',
+                        },
+                      },
+                    ],
+                  },
+                },
+                {
+                  createdAt: '2020-01-14T12:33:48.651Z',
+                  read: false,
+                  reason: 'filed_report_on_resource',
+                  from: {
+                    __typename: 'Report',
+                    id: 'reportOnComment',
+                    filed: [
+                      {
+                        reasonCategory: 'discrimination_etc',
+                        reasonDescription: 'This user is harassing me!',
+                        reportedResource: {
+                          __typename: 'Comment',
+                          id: 'c4',
+                          content: 'I am bad content in a bad comment to a bad post !!!',
+                        },
+                      },
+                    ],
+                  },
+                },
+              ]),
             },
             errors: undefined,
-          })
+          }
+
+          const response = await query({ query: notificationQuery, variables })
+          await expect(response).toMatchObject(expected)
+          await expect(response.data.notifications.length).toEqual(7) // has to be checked, because of 'arrayContaining'
         })
       })
 
       describe('filter for read: false', () => {
         it('returns only unread notifications of current user', async () => {
-          const expected = expect.objectContaining({
+          const expected = {
             data: {
               notifications: expect.arrayContaining([
                 {
+                  createdAt: '2019-08-30T19:33:48.651Z',
+                  read: false,
+                  reason: 'mentioned_in_comment',
                   from: {
                     __typename: 'Comment',
                     content: 'You have been mentioned in a comment',
                   },
-                  read: false,
-                  createdAt: '2019-08-30T19:33:48.651Z',
                 },
                 {
+                  createdAt: '2019-08-31T17:33:48.651Z',
+                  read: false,
+                  reason: 'mentioned_in_post',
                   from: {
                     __typename: 'Post',
                     content: 'You have been mentioned in a post',
                   },
+                },
+                {
+                  createdAt: '2020-01-15T16:33:48.651Z',
                   read: false,
-                  createdAt: '2019-08-31T17:33:48.651Z',
+                  reason: 'filed_report_on_resource',
+                  from: {
+                    __typename: 'Report',
+                    id: 'reportOnUser',
+                    filed: [
+                      {
+                        reasonCategory: 'discrimination_etc',
+                        reasonDescription: 'This user is harassing me with bigoted remarks!',
+                        reportedResource: {
+                          __typename: 'User',
+                          id: 'badWomen',
+                          name: 'Mrs. Badwomen',
+                        },
+                      },
+                    ],
+                  },
+                },
+                {
+                  createdAt: '2020-01-14T12:33:48.651Z',
+                  read: false,
+                  reason: 'filed_report_on_resource',
+                  from: {
+                    __typename: 'Report',
+                    id: 'reportOnComment',
+                    filed: [
+                      {
+                        reasonCategory: 'discrimination_etc',
+                        reasonDescription: 'This user is harassing me!',
+                        reportedResource: {
+                          __typename: 'Comment',
+                          id: 'c4',
+                          content: 'I am bad content in a bad comment to a bad post !!!',
+                        },
+                      },
+                    ],
+                  },
                 },
               ]),
             },
-          })
+            errors: undefined,
+          }
+
           const response = await query({
             query: notificationQuery,
             variables: { ...variables, read: false },
           })
           await expect(response).toMatchObject(expected)
-          await expect(response.data.notifications.length).toEqual(2) // double-check
+          await expect(response.data.notifications.length).toEqual(4) // has to be checked, because of 'arrayContaining'
```


see :point_up:

Reply:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2020-01-24T16:48:06Z" title="Friday, January 24th 2020, 5:48:06 pm +01:00">Jan 24, 2020</time>_

---


Same same ☝🏼